src/sagemaker/base_predictor.py

```diff
@@ -203,6 +207,7 @@ def predict(
             target_variant=target_variant,
             inference_id=inference_id,
             custom_attributes=custom_attributes,
+            target_container_hostname=target_container_hostname,
```

Comment:
It looks like this doesn't work when running something like:

```
response = runtime_sm_client.invoke_endpoint(
    EndpointName="endpoint",
    ContentType="application/json",
    TargetModel="target_model",
    TargetContainerHostname="primary",
    Body="...",
)
```

Returns:
ValidationError: An error occurred (ValidationError) when calling the InvokeEndpoint operation: Request 54a2be08-8f19-467b-aecd-6c537e45b977 endpoint does not support TargetContainerHostname